dev/tools/lib/archive_publisher.dart

```
@@ -0,0 +1,175 @@
+  final String channel;
+
+  /// The process manager to use for invoking commands. Typically only
+  /// used for testing purposes.
+  final ProcessManager processManager;
+
+  /// The temporary directory used for this publisher. If not set, one will
+  /// be created, used, and then removed automatically. If set, it will not be
+  /// deleted when done: that is left to the caller. Typically used by tests.
+  Directory tempDir;
+
+  static String gsBase = 'gs://flutter_infra';
+  static String releaseFolder = '/releases';
+  static String baseUrl = 'https://storage.googleapis.com/flutter_infra';
+  static String archivePrefix = 'flutter_';
+  static String releaseNotesPrefix = 'release_notes_';
+
+  final String metadataGsPath = '$gsBase$releaseFolder/releases.json';
+
+  /// Publishes the archive for the given constructor parameters.
+  bool publishArchive() {
+    assert(channel == 'dev', 'Channel must be dev (beta not yet supported)');
+    // Check for access early so that we don't try to publish things if the
+    // user doesn't have access to the metadata file.
+    _checkForGSUtilAccess();
+    final List<String> platforms = <String>['linux', 'mac', 'win'];
+    final Map<String, String> metadata = <String, String>{};
+    for (String platform in platforms) {
+      final String src = _builtArchivePath(platform);
+      final String dest = _destinationArchivePath(platform);
+      final String srcGsPath = '$gsBase$src';
+      final String destGsPath = '$gsBase$releaseFolder$dest';
+      _cloudCopy(srcGsPath, destGsPath);
+      metadata['${platform}_archive'] = '$channel/$platform$dest';
+    }
+    metadata['release_date'] = new DateTime.now().toUtc().toIso8601String();
+    metadata['version'] = version;
+    _updateMetadata(metadata);
+    return true;
+  }
+
+  /// Checks to make sure the user has access to the Google Storage bucket
+  /// required to publish. Will print an error and return false if not.
```

looks like this throws instead of returning false as documented?

---

Fixed.